tests/e2e/multi_vc.go

```diff
@@ -0,0 +1,1827 @@
+		defer cancel()
+		ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+		fss.DeleteAllStatefulSets(client, namespace)
+		ginkgo.By(fmt.Sprintf("Deleting service nginx in namespace: %v", namespace))
+		err := client.CoreV1().Services(namespace).Delete(ctx, servicename, *metav1.NewDeleteOptions(0))
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		framework.Logf("Perform cleanup of any left over stale PVs")
+		allPvs, err := client.CoreV1().PersistentVolumes().List(ctx, metav1.ListOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, pv := range allPvs.Items {
+			err := client.CoreV1().PersistentVolumes().Delete(ctx, pv.Name, metav1.DeleteOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		if isVsanHealthServiceStopped {
+			vCenterHostname := strings.Split(multiVCe2eVSphere.multivcConfig.Global.VCenterHostname, ",")
+			vcAddress := vCenterHostname[0] + ":" + sshdPort
+			framework.Logf("Bringing vsanhealth up before terminating the test")
+			startVCServiceWait4VPs(ctx, vcAddress, vsanhealthServiceName, &isVsanHealthServiceStopped)
+		}
+
+		if isSPSServiceStopped {
+			vCenterHostname := strings.Split(multiVCe2eVSphere.multivcConfig.Global.VCenterHostname, ",")
+			vcAddress := vCenterHostname[0] + ":" + sshdPort
+			framework.Logf("Bringing sps up before terminating the test")
+			startVCServiceWait4VPs(ctx, vcAddress, spsServiceName, &isSPSServiceStopped)
+			isSPSServiceStopped = false
+		}
+	})
+
+	/* TESTCASE-1
+
+	Stateful set with SC contains default parameters and WFC binding mode and
+	specific nodeaffinity details in statefullset
+
+	Steps:
+	1. Create SC default values so all the AZ's should be consided for provisioning.
+	2. Create Statefulset with parallel pod management policy
+	3. Wait for PVC to reach bound state and POD to reach Running state
+	4. Volumes should get distributed among the nodes which are mentioned in node affinity of
+	Statefullset yaml
+	5. Make sure common validation points are met
+	a) Verify the PV node affinity details should have appropriate node details
+	b) The Pods should be running on the appropriate nodes
+	c) CNS metadata
+	6. Scale-up /Scale-down the statefulset and verify the common validation points on newly
+	created statefullset
+	7. Clean up the data
+	*/
+
+	ginkgo.It("Workload creation on a multivc environment with sts specified with node affinity "+
+		"and SC with no allowed topology", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		parallelPodPolicy = true
+		nodeAffinityToSet = true
+		stsReplicas = 3
+		scaleUpReplicaCount = 5
+		scaleDownReplicaCount = 2
+
+		if multiVCSetupType == "multi-2vc-setup" {
+			/* here in 2-VC setup statefulset node affinity is taken as k8s-zone -> zone-1,zone-2,zone-3
+			i.e VC1 allowed topology and VC2 partial allowed topology
+			*/
+			allowedTopologyLen = 3
```

Comment:
Are we using 3 level topology lebels in k8s set up ? as i know we are using ony k8s-zone and assigning topology in all the hosts

Reply:
discussed offline with you